### PR TITLE
Add typescript

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Resolves: https://github.com/GavinJoyce/ember-headlessui/issues/18

- [x] Requires support for linting typescript
- [x] Convert at least one thing to typescript

### CODE REVIEW

File: addon/components/menu/button.hbs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{{! template-lint-disable no-down-event-binding }}
```
Comment (NullVoxPopuli) [inline addon/components/menu/button.hbs:1]: these are unrelated to typescript, but as I was debugging prettier warnings (which are reported as errors? 🤷 (I'm not used to the prettier CLI)), I decided to just take care of all warnings

File: tests/integration/components/switch-test.js [hunk 2]
```diff
@@ -164,9 +164,7 @@ module('Integration | Component | <Switch>', function (hooks) {
         assertSwitchState({ state: SwitchState.Off });
       });
     });
-  });
 
-  module('Keyboard interactions', function () {
```
Comment (GavinJoyce) [inline tests/integration/components/switch-test.js:169]: did you intend to remove this? AFAIK, this is mirroring the structure of the Vue tests

**Review by GavinJoyce — APPROVED** [review]
great!

File: tests/integration/components/switch-test.js (hunk 2 above)
Comment (NullVoxPopuli) [reply tests/integration/components/switch-test.js:169]: I did! where was a qunit lint warning about a module _with the same name_ immediately preceding, so the module separation doesn't make sense, ya know?

Comment (NullVoxPopuli) [reply tests/integration/components/switch-test.js:169]: right here: https://github.com/GavinJoyce/ember-headlessui/pull/89/files/573fa499920dd3cf91c1c1c500181eced5aa3909..d3d7b203bebde0b7f61e7b8da88056eea3a43ae3#diff-f92b87d45171cafc52ed0f458af0ac7d6779551c40dd95ea796d902a8b913b5dR140

Comment (GavinJoyce) [reply tests/integration/components/switch-test.js:169]: 👍 aha, nice warning from eslint

**Pull request merged**